### PR TITLE
DEV: Add support for the experimental user menu

### PR DESCRIPTION
**Pull request by OsamaSayegh**
This PR ports the widgets-based [implementation](https://github.com/discourse/discourse-reactions/blob/ac9b52473c177e8e9f8b7ce3fa01165bf4e61ad9/assets/javascripts/discourse/widgets/discourse-reactions-reaction-notification-item.js) of the `reaction` notification item to the experimental Glimmer-based user menu. The changes in this PR should make reaction notifications in the experimental user menu consistent with they look/behave in the old user menu.

We don't need to add a `.discourse-compatibility` entry for this change because it's fully backward compatible since we check that the `registerNotificationTypeRenderer` plugin API is available before we attempt to use it.

For more context on the experimental user menu, see https://github.com/discourse/discourse/pull/17379.

### CODE REVIEW

File: assets/javascripts/initializers/discourse-reactions.js [hunk 1]
```diff
@@ -125,6 +128,85 @@ function initializeDiscourseReactions(api) {
     // It's an admin component so it's not always present
     { ignoreMissing: true }
   );
+
+  api.replaceIcon("notification.reaction", "discourse-emojis");
+
+  if (api.registerNotificationTypeRenderer) {
+    api.registerNotificationTypeRenderer("reaction", (NotificationItemBase) => {
```
Comment (tgxworld) [inline assets/javascripts/initializers/discourse-reactions.js]:
```suggestion
    api.registerNotificationTypeRenderer("reaction", (NotificationTypeBase) => {
```

File: assets/javascripts/initializers/discourse-reactions.js [hunk 2]
```diff
@@ -125,6 +128,85 @@ function initializeDiscourseReactions(api) {
...
+        get description() {
+          if (
+            this.notification.data.count > 1 &&
+            !this.notification.data.username2
+          ) {
+            return I18n.t("notifications.reaction_1_user_multiple_posts", {
+              count: this.notification.data.count,
+            });
+          }
+          return super.description;
```
Comment (tgxworld) [inline assets/javascripts/initializers/discourse-reactions.js:205]: I'm wondering why we can't just call `super(...arguments)` here.

**Review by tgxworld — APPROVED** [review]

File: assets/javascripts/initializers/discourse-reactions.js (hunk 2 above)
Comment (OsamaSayegh) [reply assets/javascripts/initializers/discourse-reactions.js:205]: we do `super(...arguments)` in functions/methods, but for accessing properties of the parent class, we do `super.<property_name>`.

Comment (tgxworld) [reply assets/javascripts/initializers/discourse-reactions.js:205]: Oooo icic this is a getter :+1:

**Comment by OsamaSayegh** [comment]
thank you tgx!

**Pull request merged**